web/src/components/MemoContent/Link.tsx

```diff
@@ -1,18 +1,97 @@
+import { Tooltip, Card, AspectRatio, Box } from "@mui/joy";
+import { Link as MLink } from "@mui/joy";
+import { useEffect, useState } from "react";
+import { metadataServiceClient } from "@/grpcweb";
+
 interface Props {
   url: string;
   text?: string;
 }
 
 const Link: React.FC<Props> = ({ text, url }: Props) => {
+  const [title, setTitle] = useState("");
+  const [description, setDescription] = useState("");
+  const [image, setImage] = useState<string>("");
+
+  const fetchUrlMetadata = async () => {
+    try {
+      const response = await metadataServiceClient.getMetadata({ url }, {});
+
+      setTitle(response.metadata?.title || "No Title.");
+      setDescription(response.metadata?.description || "No Description.");
```

Comment:
that seems to me like a strange decision. maybe just don't show description if there isn't one?